server-common/src/main/java/org/apache/kafka/security/NoOpPasswordEncoder.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.security;
+
+import org.apache.kafka.common.config.types.Password;
+
+/**
+ * A password encoder that does not modify the given password. This is used in KRaft mode only.
+ */
+public class NoOpPasswordEncoder implements PasswordEncoder {
```

Comment:
Am wondering if this class deserve an extra file or if it can be just defined inside the PasswordEncoder interface. Maybe like this https://github.com/apache/kafka/blob/82920ffad00285c11453e64e220ac0ec2666ac54/server-common/src/main/java/org/apache/kafka/server/common/DirectoryEventHandler.java#L27

Reply:
Good idea! Done